# Settings for ros_control control loop
hardware_control_loop:
    loop_hz: 100

# Publish all joint states -----------------------
joint_state_controller:
    type: "joint_state_controller/JointStateController"
    publish_rate: 100

# Joint Trajectory Controller - position based -----------------------
niryo_one_follow_joint_trajectory_controller:
    type: "position_controllers/JointTrajectoryController"
    joints: 
        - joint_1  
        - joint_2
        - joint_3  
        - joint_4  
        - joint_5  
        - joint_6
    constraints:
        goal_time: 5.0
        stopped_velocity_tolerance: 0.05
        joint_1: {trajectory: 0.5, goal: 0.2}
        joint_2: {trajectory: 0.5, goal: 0.2}
        joint_3: {trajectory: 0.5, goal: 0.2}
        joint_4: {trajectory: 0.5, goal: 0.2}
        joint_5: {trajectory: 1.0, goal: 0.3}
        joint_6: {trajectory: 1.0, goal: 0.3}
    stop_trajectory_duration: 1.0
    state_publish_rate: 50
    # action_monitor_rate: 50

# NOTE: this requires a hardware_interface/PositionJointInterface on the gripper joints
gripper_follow_joint_trajectory_controller:
    type: "position_controllers/JointTrajectoryController"
    joints: 
       - gripper_left_finger_joint
       - gripper_right_finger_joint
    constraints:
        goal_time: 1.0
        stopped_velocity_tolerance: 0.05
        gripper_left_finger_joint: {trajectory: 0.5, goal: 0.2}
        gripper_right_finger_joint: {trajectory: 0.5, goal: 0.2}
    stop_trajectory_duration: 1.0
    state_publish_rate: 20



